clang/lib/Analysis/FlowSensitive/TypeErasedDataflowAnalysis.cpp

```diff
@@ -335,10 +326,8 @@ computeBlockInputState(const CFGBlock &Block, AnalysisContext &AC) {
         // when the terminator is taken. Copy now.
         TypeErasedDataflowAnalysisState Copy = MaybePredState->fork();
 
-        const StmtToEnvMap StmtToEnv(AC.CFCtx, AC.BlockStates);
         auto [Cond, CondValue] =
-            TerminatorVisitor(StmtToEnv, Copy.Env,
-                              blockIndexInPredecessor(*Pred, Block))
+            TerminatorVisitor(Copy.Env, blockIndexInPredecessor(*Pred, Block))
```

Comment:
I approved the PR because I believe it is a step in the right direction, but I can't help but feel like something is a bit off. I wonder if we could reorganize things a bit to make errors like this impossible to happen. Specifically, I wonder if it is possible to make sure we never need these extra analysis state copies, and we can always work on the state we have in place associated with the corresponding block. 

Here, we evaluate the side effect of the terminator stmt multiple times, once for each of the successors of the block. This is OK here, since each evaluation is happening on a separate copy of the state. While it is OK, it feels a bit redundant and error prone. I wonder if it would be possible to actually do the transfer for the terminator only once, when the basic block is processed, and here only do the `transferBranchTypeErased` without ever doing a regular `transfer`.